lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -95,6 +104,37 @@ def main():
         log.critical(f'Output file {args.output_file} already exists')
         sys.exit(1)
 
+    # read Cat-B calibration data if available
+    catB_calib = None
+    if args.catB_calibration_file is not None:
+        if Path(args.catB_calibration_file).exists():
+            log.info(f"Cat-B calbration: {args.catB_calibration_file}")
+            catB_calib = read_table(args.catB_calibration_file, "/tel_1/calibration")
+
+            # add the calibration index
+            catB_calib['calibration_id'] = np.arange(len(catB_calib))
+            catB_calib['pedestal_id'] = np.arange(len(catB_calib))
+            catB_calib['flatfield_id'] = np.arange(len(catB_calib))
+
+            catB_pedestal = read_table(args.catB_calibration_file, "/tel_1/pedestal")
+            catB_pedestal['pedestal_id'] = np.arange(len(catB_pedestal))
+
+            catB_flatfield = read_table(args.catB_calibration_file, "/tel_1/flatfield")
+            catB_flatfield['pedestal_id'] = np.arange(len(catB_flatfield))
+
+            #catB_calib_time = (np.array(catB_calib["time_max"])+np.array(catB_calib["time_min"]))/2
+            catB_calib_time = np.array(catB_calib["time_min"])
+            catB_dc_to_pe = np.array(catB_calib["dc_to_pe"])
+            catB_pedestal_per_sample = np.array(catB_calib["pedestal_per_sample"])
+
+            catB_time_correction = np.array(catB_calib["time_correction"])
+            catB_unusable_pixels = np.array(catB_calib["unusable_pixels"])
+            pixel_index = np.arange(constants.N_PIXELS)
+        else:
```

Comment:
It's often nicer to use "early exit", e.g. here, I would put the check like this:

```
if not args.catB_calibration_file.exists():
    # bad path   
    log.critical(...)
    sys.exit(1)

# happy path
log.info(f"Cat-B calbration: {args.catB_calibration_file}")
```

Reply:
Done